README.md

```diff
@@ -19,18 +19,89 @@ This code is dual licensed under the MIT license and the OGL (Open Government Li
 
 The contents of this repository are protected by Crown Copyright (C).
 
-## Development
 
-### Requirements
-* make
-* nodejs v12 or v14
-* npm/yarn
-* [poetry](https://github.com/python-poetry/poetry)
+## Setup
+
+N.B. that some functioanlity requires environment variables to be set. Some of these are described lower down in the readme, whilst others can be found in [the environment variables section of this confluence page](https://nhsd-confluence.digital.nhs.uk/display/SPINE/Personal+Demographics+Service+api+setup)
+
+
+Windows users should install [Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/install). Any distro is fine, though ubuntu/debian are recommended.
+
+## Installing requirements
+Install build requirements. This will make sure you don't hit any weird python issues later.
+```
+$ sudo apt update
+$ sudo apt install make build-essential libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev git net-tools python-openssl
+```
+If you get the error "Unable to locate package python-openssl", try
+```
+$ sudo apt install python3-openssl
+```
+
+Install [pyenv](https://github.com/pyenv/pyenv) using the code below and then follow their [guide](https://github.com/pyenv/pyenv#set-up-your-shell-environment-for-pyenv) to integrating it with your terminal
+```
+$ curl https://pyenv.run | bash
+$ exec $SHELL
+```
+If the command isn't working you can also [try the instructions here.](https://www.liquidweb.com/kb/how-to-install-pyenv-on-ubuntu-18-04/)
+
+Install python 3.8.2
+```
+$ pyenv install 3.8.2
+```
+Either set this as your global python (if this is not incompatible with your other projects),
+```
+$ pyenv global 3.8.2
+```
+or local to repository, if there is not a python-version file installed (you might have to raise a PR to add the file that's created).
+```
+$ pyenv local 3.8.2
+$ python --version
+```
+
+Install poetry, then run 'poerty install' to install dependencies. Makes sure you change directory to this repo.
+```
+$ curl -sSL https://install.python-poetry.org | python3
+$ poetry install
+```
+
+Install nvm & npn
+```
+$ curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.2/install.sh | bash
+  # Close and reopen your terminal window, or use 'exec $SHELL'
+$ nvm install lts/fermium
+$ nvm use lts/fermium
+$ npm --version
+```
+
+Install Java
+```
+$ sudo apt install default-jre default-jdk
+$ java -version
+```
+
+Install pytest
+```
+$ pip install -U pytest
+$ sudo apt-get update
+$ sudo apt-get install jq
+```
+
 
-### Install
+Next open powershell and get the wsl ip
+```
+$ wsl hostname -i
 ```
-$ make install
+Add a proxy and open the windows fire wall, replace [PORT] with the ip from the previous command
```

Comment:
putting an IP address in a `[PORT]`? Just confirming that's 100% right

Reply:
I checked and amended